python/ray/air/tests/test_data_batch_conversion.py

```diff
@@ -1,25 +1,105 @@
+        expected_output = np.array([1, 2, 3])
+        actual_output = _convert_batch_type_to_numpy(input_data)
+        np.testing.assert_array_equal(expected_output, actual_output)
+
+        input_data = np.array([1, 2, 3])
+        expected_output = input_data
+        actual_output = _convert_batch_type_to_numpy(input_data)
+        np.testing.assert_array_equal(expected_output, actual_output)
+
+    def test_arrow_to_numpy(self):
+        input_data = pa.table({"column_1": [1, 2, 3, 4]})
+        expected_output = {"column_1": np.array([1, 2, 3, 4])}
+        actual_output = _convert_batch_type_to_numpy(input_data)
+        self.assertEqual(expected_output.keys(), actual_output.keys())
+        np.testing.assert_array_equal(
+            expected_output["column_1"], actual_output["column_1"]
+        )
+
+        input_data = pa.table({TENSOR_COLUMN_NAME: [1, 2, 3, 4]})
+        expected_output = np.array([1, 2, 3, 4])
+        actual_output = _convert_batch_type_to_numpy(input_data)
+        np.testing.assert_array_equal(expected_output, actual_output)
+
+        input_data = pa.table(
+            {
+                "column_1": [1, 2, 3, 4],
+                "column_2": [1, -1, 1, -1],
+            }
+        )
+        expected_output = {
+            "column_1": np.array([1, 2, 3, 4]),
+            "column_2": np.array([1, -1, 1, -1]),
+        }
 
-    actual_output = convert_pandas_to_batch_type(actual_output, type=DataType.PANDAS)
-    pd.testing.assert_frame_equal(actual_output, input_data)
+        actual_output = _convert_batch_type_to_numpy(input_data)
+        self.assertEqual(expected_output.keys(), actual_output.keys())
+        np.testing.assert_array_equal(
+            expected_output["column_1"], actual_output["column_1"]
+        )
+        np.testing.assert_array_equal(
+            expected_output["column_2"], actual_output["column_2"]
+        )
+
+    def test_pd_dataframe_to_numpy(self):
+        input_data = pd.DataFrame({"column_1": [1, 2, 3, 4]})
+        expected_output = {"column_1": np.array([1, 2, 3, 4])}
+        actual_output = _convert_batch_type_to_numpy(input_data)
+        self.assertEqual(expected_output.keys(), actual_output.keys())
+        np.testing.assert_array_equal(
+            expected_output["column_1"], actual_output["column_1"]
+        )
+
+        input_data = pd.DataFrame(
+            {"column_1": [1, 2, 3, 4], "column_2": [1, -1, 1, -1]}
+        )
+        expected_output = {
+            "column_1": np.array([1, 2, 3, 4]),
+            "column_2": np.array([1, -1, 1, -1]),
+        }
+        actual_output = _convert_batch_type_to_numpy(input_data)
+        self.assertEqual(expected_output.keys(), actual_output.keys())
+        np.testing.assert_array_equal(
+            expected_output["column_1"], actual_output["column_1"]
+        )
+        np.testing.assert_array_equal(
+            expected_output["column_2"], actual_output["column_2"]
+        )
```

Comment:
@clarkzinzow do we also need to test if arrow->numpy and pandas->numpy paths work with the respective tensor extensions?